File: 442/find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function (nums) {
    let ans = [];
    for (let i = 0, len = nums.length; i < len; i++) {
        let item = nums[i];
        let val = Math.abs(item) - 1;

        if (nums[val] < 0) {
            ans.push(val + 1);
        } else {
            nums[val] *= -1;
        }
    }

    return ans;
};

File: 442/find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = []
        for i in range(len(nums)):
            item = nums[i]
            val = abs(item) - 1
            if nums[val] < 0:
                ans.append(val+1)
            else:
                nums[val] *= -1

        return ans
